flink-table/flink-sql-client/src/main/java/org/apache/flink/table/client/gateway/local/LocalExecutor.java

```diff
@@ -200,69 +277,98 @@ public void start() {
 	}
 
 	@Override
-	public List<String> listCatalogs(SessionContext session) throws SqlExecutionException {
-		final ExecutionContext<?> context = getOrCreateExecutionContext(session);
+	public void resetSessionProperties(String sessionId) throws SqlExecutionException {
+		// Renew the ExecutionContext by using the default environment.
+		this.contextMap.put(sessionId, createExecutionContext(defaultEnvironment));
+	}
 
-		final TableEnvironment tableEnv = context
-			.createEnvironmentInstance()
-			.getTableEnvironment();
+	@Override
+	public void setSessionProperty(String sessionId, String key, String value) throws SqlExecutionException {
+		Environment env = getExecutionContext(sessionId).getEnvironment();
+		Environment newEnv = Environment.enrich(env, ImmutableMap.of(key, value), ImmutableMap.of());
+		// Renew the ExecutionContext by merging the default environment and new environment.
+		this.contextMap.put(sessionId, createExecutionContext(defaultEnvironment, newEnv));
```

Comment:
I think we can directly call `createExecutionContext(newEnv)` here, `defaultEnvironment` should be already covered by the `env` you got from current execution context

Reply:
Sounds good, thanks for pointing it out.